### DISCUSSION TITLE
BAD GATEWAY after UPDATE

### DISCUSSION THREAD

**Discussion by kalazino** [eduardogsilva/routerfleet#45] [General]
Hello and thanks for sharing your project.
I tried to upgrade but after that I get a BAD GATEWAY.
I'm thinking of doing a new installation, but I don't know how to restore the database (I performed the backup as per procedure)
Thank you

**Comment by eduardogsilva** [comment]
You do not need to make a new installation. 
You probably just didn't updated your docker-compose.yml file. Check the [upgrade instructions](https://github.com/eduardogsilva/routerfleet?tab=readme-ov-file#upgrade-instructions-for-routerfleet) and do not skip this step.

You may also check your docker logs for the error messages.

**Reply by eduardogsilva** [reply]
@kalazino did you manage to fix your installation? 
Let me know if you need any further assitance.

Cheers!

**Reply by kalazino** [reply]
HI, 
I'll try in a few hours and let you know
Thank you

**Reply by kalazino** [reply]
I tried following the upgrade procedure step by step again, but it still doesn't work.
How can I export the logs to you?
Thank you

**Reply by eduardogsilva** [reply]
check what containers are not running with docker ps -a

You also may try docker logs -f routerfleet (or any other container that is not running)

**Reply by kalazino** [reply]
It appears that the postgres and routerfleet containers are unable to start

<img width="926" alt="image" src="https://github.com/user-attachments/assets/937605a9-69a1-4838-bbc3-fcaa0c7487c1">

**Reply by eduardogsilva** [reply]
check the logs for each container that is not starting, but it does look like that you did not upgraded your docker-compose.yml file. 

this message should be there in routerfleet container logs.

Do not skip this step, please check again the upgrade instructions.

Cheers.

**Reply by kalazino** [reply]
This is the log for container postgres

<img width="1328" alt="image" src="https://github.com/user-attachments/assets/30682bf9-7ea2-43b9-a84f-9a39f67d56dd">

I'm running docker on Alpine distro. 
could this be the problem?

**Reply by kalazino** [reply]
Postgres version 16 is specified in your docker-compose file.
I modified the file with: 
image: postgres:latest 
and it seems to work

**Reply by eduardogsilva** [reply]
I've detected the issue that the postgres:latest upgraded the current installs from 16 to 17. This was breaking the container, so I locked the version to 16.

Since you  have a "new install", you got the opposite problem.
I have to think how I'm going to deal with this now.

**Comment by eduardogsilva** [comment]
I've created an issue with this bug. I decided to pin the version 16 since I believe that this is  the most used version at this moment.

https://github.com/eduardogsilva/routerfleet/issues/46
